File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-6.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* newHead = reverse(head);
        ListNode* curr = newHead;
        if (n == 1) {
            newHead = newHead->next;
            return reverse(newHead);
        }
        for (int i = 0; i < n - 2; i++) {
            curr = curr->next;
        }
        curr->next = curr->next->next;
        return reverse(newHead);
    }

    ListNode* reverse(ListNode* head) {
        ListNode* curr = head;
        ListNode* prev = nullptr;

        while (curr) {
            ListNode* temp = curr->next;
            curr->next = prev;
            prev = curr;
            curr = temp;
        }
        return prev;
    }
};

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-6.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)

        if n == 0:
            return 0

        prev = [0] * n

        for i in range(1, n + 1):
            curr = [0] * n

            for j in range(0, n - i + 1):
                curr[j] = prev[j] + nums[i + j - 1]

                if curr[j] >= target:
                    return i

            prev = curr
        return 0